src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSWebWorker.cs

```diff
@@ -64,7 +65,17 @@ private static Task<T> RunAsyncImpl<T>(Func<Task<T>> body, CancellationToken can
                         return;
                     }
 
-                    JSHostImplementation.InstallWebWorkerInterop(false);
+                    var synchronizationContext = JSHostImplementation.InstallWebWorkerInterop(false, cancellationToken);
+
+                    reg = cancellationToken.Register(() =>
```

Comment:
It's not clear to me what this does, a brief comment would be good

Reply:
note there are 2 almost identical methods. One is generic and the other one is not. It defeated my attempts for code reuse.

Reply:
I was able to do that now. There is new `JSWebWorkerInstance<T>` which does that.